src/useObservable.ts

```diff
@@ -33,24 +38,40 @@ export function useObservable<T>(observable: Observable<T>): T | undefined
 export function useObservable<T>(observable: Observable<T>, initialValue: T): T
 export function useObservable<T>(observable: Observable<T>, initialValue: () => T): T
 export function useObservable<T>(observable: Observable<T>, initialValue?: T | (() => T)) {
-  const [getSnapshot, subscribe] = useMemo(() => {
+  const [getSnapshot, subscribe] = useMemo<
+    [() => T, Parameters<typeof useSyncExternalStore>[0]]
+  >(() => {
     const record = getOrCreateStore(observable, getValue(initialValue))!
     return [
       function getSnapshot() {
+        // @TODO: perf opt opportunity: we could do `record.subscription.unsubscribe()` here to clear up some memory, as this subscription is only needed to provide a sync initialValue.
         return record.currentValue
       },
-      function subscribe(callback: (value: T) => void) {
-        const sub = record.observable.subscribe(next => callback(next))
+      function subscribe(callback: () => void) {
+        // @TODO: perf opt opportunity: we could do `record.subscription.unsubscribe()` here as we only need 1 subscription active to keep the observer alive
+        const sub = record.observable.subscribe(() => callback())
```

Comment:
The typing for `Parameters<typeof useSyncExternalStore>[0]`:
```tsx
(onStoreChange: () => void) => () => void
```
Which is why the subscriber is rewritten to stop passing `next`.
I didn't refactor to giving it `callback`:
```suggestion
        const sub = record.observable.subscribe(callback)
```
as I don't know enough on how `rxjs` works to verify if this might cause regressions.

Reply:
Should be fine, assuming callback ignores any values passed to it